src/main.c

```diff
@@ -16,47 +16,59 @@ int main(int argc, char **argv)
     mdsys_t sys;
     double t_start;
 
-    printf("LJMD version %3.1f\n", LJMD_VERSION);
+    #ifdef _MPI
+    MPI_Init(NULL, NULL);
+    #endif
+    fill_mpi_struct(&sys);
+    
+    if (sys.mpirank==0) printf("LJMD version %3.1f\n", LJMD_VERSION);
 
     t_start = wallclock();
 
     /* read input file */
-    add_data(stdin, &line, &restfile, &trajfile, &ergfile, &sys, &nprint);
-    
+    if (sys.mpirank==0) add_data(stdin, &line, &restfile, &trajfile, &ergfile, &sys, &nprint);
```

Comment:
I think we can print here communication time, just to have idea

Reply:
Why you did add_data again here

Reply:
I just putted the if in front